### PR TITLE
Fix hdf5 output

### PR DESCRIPTION
**Pull request by gassmoeller**
It looks like our HDF5 output was broken since the conversion to solution/ subdirectories, because the paths in the .xdmf file were added twice ("solution/output_directory/solution/filename.h5") and nobody noticed. Does not seem to be our usual output format :smile:. Anyway, this fixes the paths, and adds a test for the output format (sadly another binary output test, but it is small). I might also remove the .h5 files if we prefer to keep the binary files out of the repository.

### CODE REVIEW

File: source/postprocess/visualization.cc [hunk 1]
```diff
@@ -446,30 +446,30 @@ namespace aspect
       if (output_format=="hdf5")
         {
           XDMFEntry new_xdmf_entry;
-          std::string     h5_solution_file_name = this->get_output_directory() + "solution/" + solution_file_prefix + ".h5";
-          std::string     xdmf_filename = this->get_output_directory() + "solution.xdmf";
+          const std::string h5_solution_file_name = "solution/" + solution_file_prefix + ".h5";
```
Comment (tjhei) [inline source/postprocess/visualization.cc]: I am unable to see from this patch, what actually changed, is it that ``h5_solution_file_name`` doesn't have the output directory in there?

**Comment by tjhei** [comment]
My fault, sorry. I thought I tested reading it in though, weird. :-(

It looks like we didn't have a single hdf5 test before, right? Thanks for tackling this.

File: source/postprocess/visualization.cc (hunk 1 above)
Comment (gassmoeller) [inline source/postprocess/visualization.cc]: The real fix is down in line 468. Previously we added the "solution/" string twice for the xdmf entry (one time in this line and one time down there). I just sorted creating the filenames around to make it consistent that: 

- None of the "*filename" variables contain the output_directory anymore
- All of them contain the "solution/" prefix already (if necessary)

**Comment by gassmoeller** [comment]
Not sure what happened, might also have been me when I changed the parallel output options around. Anyway, the test should now make sure that such things do not happen again.

**Comment by tjhei** [comment]
> I might also remove the .h5 files if we prefer to keep the binary files out of the repository.

I think there is not much value in having them and it will likely make the test break on other machines (different hdf5 version, rounding, etc.).

**Comment by tjhei** [comment]
(looks good otherwise, please remove the .h5 files and merge yourself)

**Pull request merged**